File: java/insertionSort.java
import java.io.*;
import java.util.Scanner;

public class insertionSort {
    public static void insertionSort(int[] array) {
        for (int i = 1; i < array.length; i++) {
            int current = array[i];
            int j = i - 1;
            while (j >= 0 && current < array[j]) {
                array[j + 1] = array[j];
                j--;
            }
            array[j + 1] = current;
        }
    }

    public static void main(String[] args) throws IOException {
        final int N = 10000;
        File array = new File("array.txt");
        Scanner scanner = new Scanner(array);
        int [] arr = new int [N];
        int i = 0;

        while(scanner.hasNextInt()){
            arr[i++] = scanner.nextInt();
        }
        final double startTime = System.nanoTime() / 1000000.0;
        insertionSort(arr);
        final double endTime = System.nanoTime() / 1000000.0;

        System.out.println((endTime - startTime));
    }
}

File: kt/insertionSort.kt
import java.util.*
import java.io.File

fun insertion_sort(arr: IntArray): Unit{
    for (i in 1 until arr.size){
        val current = arr[i]
        var j = i - 1
        while(j >= 0 && current < arr[j]){
            arr[j + 1] = arr[j]
            j--
        }
        arr[j + 1] = current
    }
}

fun main(args : Array<String>){

    val N = 10000
    val arrayContent = File("array.txt")
    var scanner = Scanner(arrayContent)
    val arr = IntArray(N)
    var i = 0

    while (scanner.hasNextInt()) {
        arr[i++] = scanner.nextInt()
    }

    val time = System.nanoTime();
    insertion_sort(arr)

    val timeMilli = (System.nanoTime() - time) / 1000000.0f

    println("$timeMilli")
}
